libs/ml/include/ml/dataloaders/word2vec_loaders/vocab.hpp

```diff
@@ -37,6 +48,65 @@ class Vocab
   SizeType    IndexFromWord(std::string const &word) const;
 };
 
+/**
+ * reserve numerical max of SizeType for unknown word
+ *
+ */
+Vocab::Vocab(){}
+
+/**
+ * Get the total count of the vocabulary
+ */
+void Vocab::Update() {
```

Comment:
this is too generic a function name. it does two (apparently unrelated) things; maybe just make this one called UpdateCount, and separately just update the reverse_data_ outside of this function